bson/_cbsonmodule.c

```diff
@@ -455,25 +479,34 @@ static long _type_marker(PyObject* object) {
  */
 int convert_codec_options(PyObject* options_obj, void* p) {
     codec_options_t* options = (codec_options_t*)p;
+    PyObject* type_registry_obj = NULL;
     long type_marker;
+
     options->unicode_decode_error_handler = NULL;
-    if (!PyArg_ParseTuple(options_obj, "ObbzO",
+
+    if (!PyArg_ParseTuple(options_obj, "ObbzOO",
                           &options->document_class,
                           &options->tz_aware,
                           &options->uuid_rep,
                           &options->unicode_decode_error_handler,
-                          &options->tzinfo)) {
+                          &options->tzinfo,
+                          &type_registry_obj))
         return 0;
-    }
 
-    type_marker = _type_marker(options->document_class);
-    if (type_marker < 0) return 0;
+    if ((type_marker = _type_marker(options->document_class)) < 0)
```

Comment:
Can you move this back outside the if statement. I think this style of assignment-if-statement makes the code harder to read and more error prone.